sdk/keyvault/azure-keyvault-secrets-apis/azure/keyvault/secrets/_shared/client_base.pyi

```diff
@@ -0,0 +1,17 @@
+# Stubs for azure.keyvault.secrets._shared.client_base (Python 3)
+#
+# NOTE: This dynamically typed stub was automatically generated by stubgen.
+
+from .._user_agent import USER_AGENT
+from ._generated import KeyVaultClient
+from .challenge_auth_policy import ChallengeAuthPolicy
+from azure.core.credentials import TokenCredential
+from azure.core.pipeline.transport import HttpTransport
+from typing import Any, Optional
+
+KEY_VAULT_SCOPE: str
+
+class KeyVaultClientBase:
+    def __init__(self, vault_endpoint: str, credential: TokenCredential, transport: Optional[HttpTransport]=..., api_version: Optional[str]=..., **kwargs: Any) -> None: ...
```

Comment:
Is the chained default credential a `TokenCredential`? Are we constraining the `credential` parameter too much?

Reply:
`TokenCredential` is the Protocol supported by all credentials, including chained credentials and `DefaultAzureCredential`.

Reply:
Presumably something like shared access or api key credentials do not provide a get_token method?

Reply:
Key Vault only supports AAD authentication